lib/Runtime/Types/ScriptFunctionType.cpp

```diff
@@ -31,7 +31,7 @@ namespace Js
             scriptContext, functionPrototype,
             address,
             proxy->GetDefaultEntryPointInfo(),
-            library->ScriptFunctionTypeHandler(proxy->IsLambda() || proxy->IsAsync() || proxy->IsClassMethod(), proxy->GetIsAnonymousFunction()),
+            library->ScriptFunctionTypeHandler(!proxy->IsConstructor() || proxy->IsAsync() || proxy->IsClassMethod(), proxy->GetIsAnonymousFunction()),
```

Comment:
`IsAsync` and `IsClassMethod` both imply `!IsConstructor` so are also redundant and should also be removed.